src/jllvm/vm/VirtualMachine.cpp

```diff
@@ -246,6 +238,32 @@ jllvm::VirtualMachine::VirtualMachine(BootOptions&& bootOptions)
                       executeObjectConstructor(root, "(Ljava/lang/String;)V", string);
                       return root;
                   }},
+        std::pair{"jllvm_build_unsatisfied_link_error",
+                  [&](ClassObject* classObject, Method* method) -> Object*
+                  {
+                      MethodType methodType = method->getType();
+
+                      llvm::StringRef methodName = method->getName();
+                      std::string className = FieldType{classObject->getDescriptor()}.pretty();
+                      std::string returnType = methodType.returnType().pretty();
+                      std::string paramTypes;
+
+                      for (auto param = methodType.param_begin(); param != methodType.param_end();)
+                      {
+                          paramTypes.append((*param).pretty());
+                          if (++param != methodType.param_end())
+                          {
+                              paramTypes.append(", ");
+                          }
+                      }
```

Comment:
I think this likely reads nicer if you use `llvm::interleave` or even `llvm::interleaveComma` here instead.